tools/tplg_parser/tplg_parser.c

```diff
@@ -441,7 +441,7 @@ int tplg_load_one_control(struct snd_soc_tplg_ctl_hdr **ctl, char **priv_data,
 	ctl_hdr = (struct snd_soc_tplg_ctl_hdr *)malloc(hdr_size);
 	if (!ctl_hdr) {
 		fprintf(stderr, "error: mem alloc\n");
-		return -EINVAL;
+		return -errno;
```

Comment:
Need to be careful here as some calls dont set errno. It's safe to return -ENOMEM for calloc and friends though.

Reply:
We know that malloc/calloc will set errno when they return NULL. Sure, care needs to be taken that we only rely on guaranteed behavior in the case of other functions.

Reply:
I checked from Ubuntu manpages that both fseek() and alloc() set errno.